lib/execution_engine2/utils/MongoUtil.py

```diff
@@ -146,13 +146,27 @@ def get_job_log(self, job_id=None) -> JobLog:
         if job_id is None:
             raise ValueError("Please provide a job id")
         with self.mongo_engine_connection():
-            return JobLog.objects.with_id(job_id)
+            try:
+                job_log = JobLog.objects.with_id(job_id)
+            except Exception:
+                raise ValueError(
+                    "Unable to find job:\nError:\n{}".format(traceback.format_exc())
+                )
+
+        return job_log
 
     def get_job(self, job_id=None) -> Job:
         if job_id is None:
             raise ValueError("Please provide a job id")
         with self.mongo_engine_connection():
-            return Job.objects.with_id(job_id)
+            try:
+                job = Job.objects.with_id(job_id)
+            except Exception:
+                raise ValueError(
+                    "Unable to find job:\nError:\n{}".format(traceback.format_exc())
```

Comment:
If you think we should raise exceptions here for being unable to find a job due to supplying a bad objectid, then we should also raise an exception if the job is None in case the objectID supplied is OK, but that job doesn't exist

Reply:
👍  good point.

Reply:
Sorry this is a typo, "if the job IS none" we should throw an exception